### DISCUSSION TITLE
How does minestom handle entities and chunks.

### DISCUSSION THREAD

**Discussion by 1k3w** [Minestom/Minestom#1363] [Help] [answered]
My question is: Does minestom track every entity regardless if it's in a loaded chunk, or does it only track entities that are in loaded chunks, and clears the ones that are not in loaded chunks.

Basically: do i save all the entities on shutdown, or do i save them when a chunk unloads.

Thanks!

**Comment by mworzala** [comment, marked as answer]
Entities are handled by an `EntityTracker` at the `Instance` level. Looking into[ the implementation of `Instance#unloadChunk` in `InstanceContainer`](https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/instance/InstanceContainer.java#L227), we can see that all entities in the chunk are removed when the chunk is unloaded.
